meetings/working-groups/extensions/metadata-names.md

```diff
@@ -0,0 +1,185 @@
+Given that only the compiler and docIDs make use of grouping and marker types, there would be no binary breaking change. Only implementation methods are referenced in IL, and those are unaffected by the change.  
+
+# Alternative proposals
+
+We also brainstormed a design where docIDs for extensions would be produced by taking `ExtensionGroupingName` and `ExtensionMarkerName`, would strip any arity suffix (to deal with metadata from another tool), then would proceed as usual (add an arity suffix.
+
+# References
+Some pointers to roslyn codebase for reference:
+
+## Producing docIDs
+All implementations use `Name` from the symbol (as opposed to `MetadataName`) and append a suffix for generics (either an arity suffix for docIds or type arguments for reference Ids on constructed symbols)
+
+1. `ISymbol.GetDocumentationCommentId()` (implemented as part of C# and VB symbols)
+This API also produces the docIDs emitted in xml output (see `DocumentationCommentCompiler.WriteDocumentationCommentXml` and `GenerateDocumentationComments`)
+Uses `Name`, either appends an arity suffix (for definitions) or type arguments (for constructed symbols). (see `DocumentationCommentIDVisitor.PartVisitor.VisitNamedType`)
+
+2. `DocumentationCommentId.CreateDeclarationId(ISymbol)` (implemented in core compiler layer)
+Uses `Name`, appends an arity suffix (see `DocumentationCommentId.PrefixAndDeclarationGenerator.DeclarationGenerator.VisitNamedType`)
+
+3. `DocumentationCommentId.CreateReferenceId()` (implemented in core compiler layer)
+Uses Name (see `BuildDottedName`), either appends an arity suffix or type arguments (see `DocumentationCommentId.ReferenceGenerator`)
+
+## Reading docIDs
+1. `DocumentationCommentId.GetSymbolsForDeclarationId(string id, Compilation compilation)`
+`DocumentationCommentId.Parser.ParseDeclaredId`  
+Splits identifier into name and arity, then searches symbols with matching Name and Arity (exact).
+
+2. CREF binding
+`BindNameMemberCref`, `ComputeSortedCrefMembers`  
+Takes a name followed by type argument list (which provides arity), looks up by name and arity (exact, for types), then constructs with type arguments
+
+
+## Metadata names and metadata loading
+The C# and VB compilers generally appends the arity suffix for generic types, to make the metadata name for a named type. There's some exceptions (notably EE named type symbols).
+For extension grouping and marker types, we use the grouping and marker names as-is, without appending a suffix. The grouping and marker names are the metadata names.
+
+When loading types from metadata, the C# and VB compilers remove the arity suffix when it matches the arity found in metadata (see `MetadataHelpers.UnmangleMetadataNameForArity`).
+
+This means that we can load a type whose name doesn't include a suffix.
+
```

Comment:
Current compiler's behavior is a proof of that because right now it has to load grouping types without suffix.